### PR TITLE
Add "server_name" parameter to the join room request

### PR DESCRIPTION
**Pull request by bmarty**
Related to vector-im/riot-android#3204

### Pull Request Checklist

* [ ] Pull request is based on the develop branch
* [ ] Pull request updates [CHANGES.rst](https://github.com/matrix-org/matrix-android-sdk/blob/develop/CHANGES.rst)
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

File: matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java [hunk 1]
```diff
@@ -1417,31 +1417,44 @@ public void onSuccess(Void info) {
...
      */
     public void joinRoom(String roomIdOrAlias, final ApiCallback<String> callback) {
+        joinRoom(roomIdOrAlias, null, callback);
+    }
+
+    /**
+     * Join a room by its roomAlias
+     *
+     * @param roomIdOrAlias the room alias
+     * @param callback      the async callback once the room is joined. The RoomId is provided.
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java]: Doc for `viaServers` is missing.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/RoomsApi.java [hunk 2]
```diff
@@ -146,23 +147,17 @@
...
      * Join the room with a room id or an alias.
      *
      * @param roomAliasOrId a room alias (or room id)
+     * @param viaServers    The servers to attempt to join the room through. One of the servers must be participating in the room.
      * @param params        the extra join param
      */
     @POST("join/{roomAliasOrId}")
-    Call<RoomResponse> joinRoomByAliasOrId(@Path("roomAliasOrId") String roomAliasOrId, @Body Map<String, Object> params);
+    Call<RoomResponse> joinRoomByAliasOrId(@Path("roomAliasOrId") String roomAliasOrId,
+                                           @Query("server_name") List<String> viaServers,
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/RoomsApi.java]: Does this annotation create queries like `server_name=server1.com&server_name=server2.com` when they are several `viaServers`?

Comment (bmarty) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/RoomsApi.java]: Yes, see https://square.github.io/retrofit/2.x/retrofit/index.html?retrofit2/http/Query.html
"Passing a List or array will result in a query parameter for each non-null item."

File: matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java (hunk 1 above)
Comment (bmarty) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java]: Done, thanks

**Pull request merged**